torch/_dynamo/variables/torch.py

```diff
@@ -781,8 +781,8 @@ def speculate_branch(branch):
                 unimplemented(true_cmp.diff(false_cmp))
 
             # Add guards
-            tx.output.guards |= false_guards
-            tx.output.guards |= true_guards
+            tx.output.tracing_context.guards_context.dynamo_guards |= false_guards
+            tx.output.tracing_context.guards_context.dynamo_guards |= true_guards
```

Comment:
Why not use guards prop

Reply:
`AttributeError: can't set attribute 'guards'` Not sure this is allowed w/ props?

Reply:
oh, I guess you need to do tx.output.guards.update(*false_guards), to avoid assignment

Reply:
This feels a little better, but I dont know - is there a reason for preferring one to the other?